kaldb/src/main/java/com/slack/kaldb/chunk/Chunk.java

```diff
@@ -24,4 +26,6 @@
 
   /** Return true if the chunk contains data within that time range (epoch ms). */
   boolean containsDataInTimeRange(long startTs, long endTs);
+
+  Map<String, FieldType> getSchema();
```

Comment:
Hmm! I'm not a big fan of `Map<String, FieldType>`

In the current codebase, we expose a ChunkSchema as `Map<String, LuceneFieldDef>`

Thoughts on keeping it conistent with that and then in the mapping API extracting the fieldtype from it?

Reply:
I didn't like using `LuceneFieldDef`, as the object returned via this API needs to be datasource agnostic. `LuceneFieldDef` today is mostly a wrapper on the `FieldType` that includes Lucene specific constructs (ie `storeDocValue`).

I think we may need another object type here that we could return the `FieldType` with instance-specific options (ie, date fields with different formatting). This abstraction currently does not exist however, so the `FieldType` is the best option we have today.

Reply:
yeah agreed! We should not expose LuceneFieldDef but FieldType needs could be extended in the future to contain more attributes.